sys/arm64/arm64/elf_machdep.c

```diff
@@ -116,6 +127,158 @@ static struct sysentvec elf64_freebsd_sysvec = {
+static struct sysentvec elf64cb_freebsd_sysvec = {
+	.sv_size	= SYS_MAXSYSCALL,
+	.sv_table	= sysent,
+	.sv_fixup	= __elfN(freebsd_fixup),
+	.sv_sendsig	= sendsig,
+	.sv_sigcode	= freebsd64cb_sigcode,
+	.sv_szsigcode	= &freebsd64cb_szsigcode,
+	.sv_name	= "FreeBSD ELF64CB",
+	.sv_coredump	= __elfN(coredump),
+	.sv_elf_core_osabi = ELFOSABI_FREEBSD,
+	.sv_elf_core_abi_vendor = FREEBSD_ABI_VENDOR,
+	.sv_elf_core_prepare_notes = __elfN(prepare_notes),
+	.sv_minsigstksz	= MINSIGSTKSZ,
+	.sv_minuser	= VM_MIN_ADDRESS,
+	.sv_maxuser	= VM_MAXUSER_ADDRESS,
+	.sv_usrstack	= USRSTACK,
+	.sv_psstringssz	= sizeof(struct ps_strings),
+	.sv_stackprot	= VM_PROT_RW_CAP,
+	.sv_copyout_auxargs = __elfN(freebsd_copyout_auxargs),
+	.sv_copyout_strings = exec_copyout_strings,
+	.sv_setregs	= exec_setregs,
+	.sv_fixlimit	= NULL,
+	.sv_maxssiz	= NULL,
+	.sv_flags	= SV_SHP | SV_TIMEKEEP | SV_ABI_FREEBSD | SV_LP64 |
+	    SV_RNG_SEED_VER | SV_CHERI | SV_UNBOUND_PCC,
+	.sv_set_syscall_retval = cpu_set_syscall_retval,
+	.sv_fetch_syscall_args = cpu_fetch_syscall_args,
+	.sv_syscallnames = syscallnames,
+	.sv_shared_page_base = SHAREDPAGE,
+	.sv_shared_page_len = PAGE_SIZE,
+	.sv_schedtail	= NULL,
+	.sv_thread_detach = NULL,
+	.sv_trap	= NULL,
+	.sv_hwcap	= &elf_hwcap,
+	.sv_hwcap2	= &elf_hwcap2,
+	.sv_onexec_old	= exec_onexec_old,
+	.sv_onexit	= exit_onexit,
+	.sv_regset_begin = SET_BEGIN(__elfN(regset)),
+	.sv_regset_end	= SET_LIMIT(__elfN(regset)),
+};
+INIT_SYSENTVEC(elf64cb_sysvec, &elf64cb_freebsd_sysvec);
+#ifdef CHERI_CAPREVOKE
+SYSINIT(caprevoke_elf64cb, SI_SUB_VM, SI_ORDER_ANY, caprevoke_sysvec_init,
+    &elf64cb_freebsd_sysvec);
+#endif
+
+static __ElfN(Brandinfo) elf64cb_freebsd_brand_info = {
+	.brand		= ELFOSABI_FREEBSD,
+	.machine	= EM_AARCH64,
+	.compat_3_brand	= "FreeBSD",
+	.interp_path	= "/libexec/ld-elf.so.1",
+	.sysvec		= &elf64cb_freebsd_sysvec,
+	.interp_newpath	= "/libexec/ld-elf64cb.so.1",
+	.brand_note	= &__elfN(freebsd_brandnote),
+	.flags		= BI_CAN_EXEC_DYN | BI_BRAND_NOTE,
+#if __has_feature(capabilities)
+	.header_supported = &elf64cb_header_supported,
+#endif
+};
+
+SYSINIT(elf64cb, SI_SUB_EXEC, SI_ORDER_FIRST,
+    (sysinit_cfunc_t)__elfN(insert_brand_entry), &elf64cb_freebsd_brand_info);
+
+static Elf_Note benchmark_abi_note = {
+	.n_namesz = sizeof(ELF_NOTE_CHERI),
+	.n_descsz = sizeof(uint32_t),
+	.n_type = NT_CHERI_MORELLO_PURECAP_BENCHMARK_ABI
+};
+
+static bool
+benchmark_abi_note_cb(const Elf_Note *note, void *arg0, bool *res)
+{
+	uint32_t *arg;
+	const char *p;
+
+	arg = arg0;
+	p = (const char *)(note + 1);
+	p += roundup2(note->n_namesz, 4);
+	*arg = *(const uint32_t *)p;
+	*res = true;
+	return (true);
+}
+
+static bool
+get_benchmark_abi_note(const struct image_params *imgp, uint32_t *res)
+{
+	const __ElfN(Phdr) *phdr;
+	const __ElfN(Ehdr) *hdr;
+	int i;
+
+	hdr = (const Elf_Ehdr *)imgp->image_header;
+	phdr = (const Elf_Phdr *)(imgp->image_header + hdr->e_phoff);
+	for (i = 0; i < hdr->e_phnum; i++)
```

Comment:
Shouldn't it use `{}` here and in other places (sys/cheri/cheri_exec.c, sys/arm64/arm64/vm_machdep.c) for multi-line statements?

Reply:
There’s a single statement, an if statement

Reply:
I'm fine with that but I still wonder what's the official style(9) here since it says:
> Two styles of braces (‘{’ and ‘}’) are allowed for single line statements.  Either they are used for all single statements, or they are used only where needed for clarity.
The first sentence mentions a single line while the second only mentions a single statement.

Reply:
This is allowed and brackets would be too for clarity. I find it adequately readable especially with the blank line before the `return (false);`